feat(node-sdk): add support for multiple metric readers

nidhiisinghh opened the pull request:
# feat(node-sdk): add support for multiple metric readers (metricReaders), deprecate metricReader, and add tests

<!--
We appreciate your contribution to the OpenTelemetry project!

Before creating a pull request, please make sure:
- Your PR is solving one problem
- Please provide enough information so that others can review your pull request
- You have read the guide for contributing
  - See https://github.com/open-telemetry/opentelemetry-js/blob/main/CONTRIBUTING.md
- You signed all your commits (otherwise we won't be able to merge the PR)
  - See https://github.com/open-telemetry/community/blob/main/guides/contributor#sign-the-cla
- You added unit tests for the new functionality
- You mention in the PR description which issue it is addressing, e.g. "Fixes #xxx". This will auto-close
  the issue that your PR fixes (if such)
-->

## Which problem is this PR solving?

The NodeSDK constructor currently only supports a single `metricReader` option, which limits users to configuring only one metric reader at a time. This is inconsistent with other options like `spanProcessors` and `logRecordProcessors` which support multiple instances.

Users who need to export metrics to multiple destinations (e.g., both console and Prometheus) currently have no clean way to do this through the NodeSDK configuration. They would need to manually create a MeterProvider and register multiple readers, which bypasses the convenience of the NodeSDK.

This PR adds support for multiple metric readers through a new `metricReaders` (plural) option while maintaining backward compatibility by deprecating the old `metricReader` (singular) option.

Fixes #5760

## Short description of the changes

### Added Features:
- **New `metricReaders` option**: Added `metricReaders?: IMetricReader[]` to `NodeSDKConfiguration` interface
- **Multiple metric reader support**: NodeSDK now accepts an array of metric readers and registers them all with the MeterProvider
- **Backward compatibility**: The old `metricReader` option still works but shows a deprecation warning
- **Comprehensive testing**: Added tests for multiple metric readers, deprecation warnings, and edge cases

### Implementation Details:
- **Interface Update**: Added `metricReaders?: IMetricReader[]` to `NodeSDKConfiguration` in `types.ts`
- **Constructor Logic**: Modified `NodeSDK` constructor to handle both `metricReaders` and `metricReader` options
- **Deprecation Warning**: Added warning message when using the old `metricReader` option
- **MeterProvider Integration**: Properly passes all metric readers to the MeterProvider constructor

### Files Changed:
- `experimental/packages/opentelemetry-sdk-node/src/types.ts`: Added `metricReaders` option to interface
- `experimental/packages/opentelemetry-sdk-node/src/sdk.ts`: Updated constructor logic to handle multiple readers
- `experimental/packages/opentelemetry-sdk-node/test/sdk.test.ts`: Added comprehensive test coverage
- `examples/multiple-metric-readers.js`: Added working example demonstrating the new functionality
- `examples/multiple-metric-readers-README.md`: Added documentation for the new feature

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

### Test Coverage Added:
1. **Multiple Metric Readers Test**: Verifies that multiple metric readers are properly registered with the MeterProvider
2. **Deprecation Warning Test**: Ensures the deprecation warning is shown when using the old `metricReader` option
3. **No Deprecation Warning Test**: Confirms no deprecation warning is shown when using the new `metricReaders` option

### Test Configuration:
- All tests run with `npm test` in the `experimental/packages/opentelemetry-sdk-node` directory
- Tests use both `ConsoleMetricExporter` and `InMemoryMetricExporter` to verify different reader types
- Environment variables are properly cleaned up after each test
- Sinon spies are used to verify deprecation warnings

### Manual Testing:
- Verified the example in `examples/multiple-metric-readers.js` works correctly
- Confirmed backward compatibility with existing `metricReader` usage
- Tested that environment variable configuration still works as expected

### Test Results:
- All 76 existing tests pass (no regressions)
- 3 new tests pass for the new functionality
- Linting passes with `npm run lint`
- Code coverage remains high (95.01% statements, 90.97% branches)

## Checklist:

- [x] Followed the style guidelines of this project
- [x] Unit tests have been added
- [x] Documentation has been updated

## Usage Examples

### New API (Recommended):
```javascript
const sdk = new NodeSDK({
  metricReaders: [
    new PeriodicExportingMetricReader({
      exporter: new ConsoleMetricExporter(),
      exportIntervalMillis: 1000,
    }),
    new PrometheusExporter({
      port: 9464,
      endpoint: '/metrics',
    }),
  ],
});
```

### Old API (Deprecated):
```javascript
const sdk = new NodeSDK({
  metricReader: new PeriodicExportingMetricReader({
    exporter: new ConsoleMetricExporter(),
  }),
  // Shows deprecation warning: "The 'metricReader' option is deprecated. Please use 'metricReaders' instead."
});
```

## Migration Guide

Users currently using the `metricReader` option should migrate to `metricReaders`:

```javascript
// Before
const sdk = new NodeSDK({
  metricReader: myMetricReader,
});

// After
const sdk = new NodeSDK({
  metricReaders: [myMetricReader],
});
```

The old option will continue to work but will show a deprecation warning. It's recommended to migrate for future compatibility.

## Breaking Changes

None. This is a non-breaking change that adds new functionality while maintaining backward compatibility.

pichlermarc commented on experimental/packages/opentelemetry-sdk-node/test/sdk.test.ts:
```diff
@@ -322,6 +322,131 @@ describe('Node SDK', () => {
+      sdk.start();
+
+      // Verify deprecation warning was shown
+      const metricReaderDeprecationWarnings = warnSpy
+        .getCalls()
+        .filter(
+          call =>
+            call.args[0] ===
+            "The 'metricReader' option is deprecated. Please use 'metricReaders' instead."
+        );
+      assert.strictEqual(metricReaderDeprecationWarnings.length, 1);
```
We can use sinon's assert functions to get more meaningful errors.

```suggestion
      sinon.assert.calledWith(
        warnSpy,
        "The 'metricReader' option is deprecated. Please use 'metricReaders' instead."
      );
```

pichlermarc commented on experimental/packages/opentelemetry-sdk-node/test/sdk.test.ts:
```diff
@@ -322,6 +322,131 @@ describe('Node SDK', () => {
+      sdk.start();
+
+      // Verify no metricReader deprecation warning was shown
+      const metricReaderDeprecationWarnings = warnSpy
+        .getCalls()
+        .filter(
+          call =>
+            call.args[0] ===
+            "The 'metricReader' option is deprecated. Please use 'metricReaders' instead."
+        );
+      assert.strictEqual(metricReaderDeprecationWarnings.length, 0);
```
We can use sinon's assert functions to get some more meaningful errors.
```suggestion
      sinon.assert.neverCalledWith(
        warnSpy,
        "The 'metricReader' option is deprecated. Please use 'metricReaders' instead."
      );
```

nidhiisinghh commented:
Hey, I’ve addressed the requested changes, updated the branch, and committed the suggestions. Also added the changelog entry. Kindly review again when convenient. Thanks!

dyladan commented on examples/multiple-metric-readers-README.md:
```diff
@@ -0,0 +1,74 @@
+# Multiple Metric Readers Example
```
Looks like this was marked resolved without being addressed. I'm going to unresolve it.

pichlermarc commented on examples/multiple-metric-readers.js:
```diff
@@ -0,0 +1,78 @@
+/*
```
I don't think this example is necessary. It's a list of metric readers, that should be fairly self-explainatory. :)

pichlermarc commented:
@nidhiisinghh - thanks for updating the changelog, looks like the other two comments still need to be addressed, then we can get this PR merged :)
